app/controllers/api/v1/contacts_controller.rb

```diff
@@ -0,0 +1,31 @@
+module Api
+  module V1
+    class ContactsController < ApplicationController
+      before_action :authenticate_user
+
+      def index
+        contacts = @current_user.contacts
+        if contacts.empty?
+          render json: { data: [], message: "No contacts found" }, status: :ok
+        else
+          render json: ContactsSerializer.new(contacts), status: :ok
+        end
+      end
+
+			def create 
+				contact = @current_user.contacts.new(contact_params)
+				if contact.save
+					render json: ContactsSerializer.new(contact), status: :created
+				else
+					render json: { error: contact.errors.full_messages.to_sentence }, status: :unprocessable_entity
```

Comment:
Excellent error messages, gracefully leads user to a better resolution. You definitely went above and beyond on this functionality, I will be referencing this in my future code.